File: Laborator/CSharp/AgentieTurism/Repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AgentieTurism.Models;

namespace AgentieTurism.Repository
{
    public interface IRepository<T, ID> where T : IIdentifiable<ID>
    {
        void Add(T entity);
        void Update(ID id, T entity);
        void Delete(ID id);
        T FindOne(ID id);
        List<T> FindAll();
    }
}

File: Laborator/Java/AgentieTurism/src/main/java/ro/mpp2024/repository/IRepository.java
package ro.mpp2024.repository;

import ro.mpp2024.domain.Identifiable;

public interface IRepository<ID, T extends Identifiable<ID>> {
    void add(T elem);
    void update(ID id, T elem);
    void delete(ID id);
    T findOne(ID id);
    Iterable<T> findAll();
}